dlc-manager/src/manager.rs

```diff
@@ -711,6 +816,43 @@ where
 
         Ok(())
     }
+
+    /// Function to call when we detect that a contract was closed by our counter party.
+    /// This will update the state of the contract and return the [`Contract`] object.
+    pub fn on_counterparty_close(
+        &mut self,
+        contract: &SignedContract,
+        closing_tx: Transaction,
+        confirmations: u32,
+    ) -> Result<Contract, Error> {
+        // check if it is the refund tx (easy case)
+        if contract.accepted_contract.dlc_transactions.refund.txid() == closing_tx.txid() {
+            let refunded = Contract::Refunded(contract.clone());
+            self.store.update_contract(&refunded)?;
+            return Ok(refunded);
+        }
+
+        let contract = if confirmations < NB_CONFIRMATIONS {
```

Comment:
It might be worth checking that the transaction is actually spending the funding tx?